starknet_py/utils/typed_data.py

```diff
@@ -20,6 +23,82 @@ class Parameter:
     type: str
 
 
+class BasicType:
+    V0 = [
+        "felt",
+        "bool",
+        "selector",
+        "merkletree",
+        "string",
+    ]
+    V1 = [
+        "felt",
+        "bool",
+        "selector",
+        "merkletree",
+        "string",
+        "enum",
+        "i128",
+        "u128",
+        "ContractAddress",
+        "ClassHash",
+        "timestamp",
+        "shortstring",
+    ]
+
+    @staticmethod
+    def get_by_revision(revision: Revision) -> List[str]:
+        return BasicType.V0 if revision is Revision.V0 else BasicType.V1
+
+
+class PresetType:
+    V1 = ["u256", "TokenAmount", "NftId"]
+
+
+@dataclass
+class Domain:
+    """
+    Dataclass representing a domain object (StarkNetDomain, StarknetDomain)
+    """
+
+    name: str
+    version: str
+    chain_id: Union[str, int]
+    revision: Optional[Union[str, int]] = None
+
+    def __post_init__(self):
+        self.resolved_revision = (
+            Revision(self.revision) if self.revision else Revision.V0
+        )
+        self.separator_name = self._resolve_separator_name()
+
+    def _resolve_separator_name(self):
+        if self.resolved_revision is Revision.V0:
+            return "StarkNetDomain"
+        return "StarknetDomain"
+
+    @staticmethod
+    def from_dict(data: DomainDict) -> "Domain":
+        """
+        Create Domain dataclass from dictionary.
+
+        :param data: Domain dictionary.
+        :return: Domain dataclass instance.
+        """
+        return cast(Domain, DomainSchema().load(data))
+
+    def to_dict(self) -> dict:
+        """
+        Create Domain dictionary from dataclass.
+
+        :return: Domain dictionary.
+        """
+        domain_dict = asdict(self)
+        # Rename chain_id to chainId when converting to dictionary
+        domain_dict["chainId"] = domain_dict.pop("chain_id")
+        return domain_dict
+
+
```

Comment:
`to_dict()` has been added because when converting `Domain` instance to dict, a field `chain_id` should serialize to `chainId`. However using `asdict()` doesn't give such a result. Maybe there's any other way, without creating an extra method?

Reply:
`asdict` is a function from `dataclasses` module and it doesn't perform a conversion using a marshmallow schema here. Let's try something similar to this:
```
    return cast(Dict, DomainSchema().dump(obj=self))
```